### ISSUE TITLE
PostgreSQL 'partial' locking hint on JOIN

### ISSUE THREAD

**Issue by ademchenko** [linq2db/linq2db#2448]
As many other RDBMS PostgreSQL supports fine-grained locks. This is made with locking hints on query (which is also very similar to other database systems) like:
```
SELECT* 
FROM USER
WHERE id = 1 FOR NO KEY UPDATE;
```
tells that we are going to update some field(s) (except id) and the hint "FOR NO KEY UPDATE" will cause other writers of user with id = 1 to wait for the end of the current transaction.

As far as I understand concerning linq2db that hint could be added right before the actual query with NextQueryHints like:
`dataConnection.NextQueryHints.Add("FOR NO KEY UPDATE");`
(BTW, maybe there's a better way of doing that?)

But it's not clear how I could add a 'partial' locking hint on join when just a part of that join needs to be locked:
```
SELECT* 
FROM USER u
JOIN WEBSITE w
on u.SiteId = w.Id
where u.email = 'alex@example.com' FOR NO KEY UPDATE OF u;

```
As you can see in the above example, to be able to lock just a user part I need to use table alias (OF u) if any. linq2db generates joins with aliases (in fact, in some cases it's impossible not to use aliases) so NextQueryHints approach doesn't work since I don't know the alias name before the query.

Could you, please, help to get the partial hints worked with linq2db?

**Comment by sdanyliv** [comment]
We need API design here. There are also SQL Server and Oracle which also has a lot of hints.

**Comment by MaceWindu** [comment]
Implemented by #3297 (will be released with next v4 preview)
